File: src/app-project/ios/proshake/glitter-interface/Notification.swift
//
//  notification.swift
//  proshake
//
//  Created by Jianzhi.wang on 2023/7/9.
//

import Foundation
import Glitter_IOS
import FirebaseMessaging
class Notification{
    
    public static func createShareInterface(){
        GlitterActivity.sharedInterFace.append(JavaScriptInterFace(functionName: "regNotification", function: {
            request in
            print("註冊推播主題")
            // 註冊推播主題
            Messaging.messaging().subscribe(toTopic: "\(request.receiveValue["topic"] ?? "")") { error in
                if let error = error {
                    print("無法註冊推播主題: \(error.localizedDescription)")
                } else {
                    print("成功註冊推播主題:\(request.receiveValue["topic"] ?? "")")
                }
            }
        }))
       
        GlitterActivity.sharedInterFace.append(JavaScriptInterFace(functionName: "getFireBaseToken", function: {
            request in
            request.responseValue["token"]=AppDelegate.fireBaseToken
            request.finish()
        }))
        
        //
        
        GlitterActivity.sharedInterFace.append(JavaScriptInterFace(functionName: "deleteFireBaseToken", function: {
            request in
            Messaging.messaging().deleteData(completion: {
                error in
            })
            request.finish()
        }))

    }
    
}

File: src/app-project/android/app/src/main/java/www/smilebio/io/glitter_interface/Notification.kt
package www.smilebio.io.glitter_interface

import android.util.Log
import com.google.firebase.messaging.FirebaseMessaging
import com.jianzhi.glitter.GlitterActivity
import com.jianzhi.glitter.JavaScriptInterFace

class Notification {
    companion object{
        fun createShareInterface(){
            GlitterActivity.addJavacScriptInterFace(JavaScriptInterFace("regNotification") {
                FirebaseMessaging.getInstance().subscribeToTopic(it.receiveValue["topic"] as String)
                it.fin()
            })
            GlitterActivity.addJavacScriptInterFace(JavaScriptInterFace("getFireBaseToken") {
                try {
                    Log.e("getFireBaseToken","start")
                    FirebaseMessaging.getInstance().token
                        .addOnCompleteListener { task ->

                            if (!task.isSuccessful) {
                                Log.w("MainActivity", "Fetching FCM registration token failed", task.exception)
                                it.responseValue["token"]=""
                                it.fin()
                                return@addOnCompleteListener
                            }

                            // Get new FCM registration token
                            val token: String = task.result
                            Log.e("token","token->${token}")
                            it.responseValue["token"]=token
                            it.fin()
                        }

                }catch (e:Exception){
Log.e("getFireBaseToken",e.toString())
                }

            })
        }
    }
}